File: backend/src/main/java/net/zorphy/backend/site/catan/dto/GameState.java
package net.zorphy.backend.site.catan.dto;

import java.time.LocalDateTime;
import java.util.List;

public record GameState(
        LocalDateTime startTime,
        GameConfig gameConfig,
        int currentPlayerTurn,
        int currentShipTurn,
        List<DicePair> classicCards,
        List<Character> eventCards,
        List<DiceRoll> diceRolls
) {
}

File: frontend/src/app/dto/sites/catan/game/GameState.ts
import {DiceRoll} from "../DiceRoll";
import {GameConfig} from "./GameConfig";
import {DicePair} from "../DicePair";
import {GameStateBase} from "../../GameStateBase";

export interface GameState extends GameStateBase {
    gameConfig: GameConfig,
    currentPlayerTurn: number,
    currentShipTurn: number,
    classicCards: DicePair[],
    eventCards: string[],
    diceRolls: DiceRoll[]
}